Modules/_abc.c

```diff
@@ -0,0 +1,1049 @@
+    0,                                  /*tp_setattr*/
+    0,                                  /*tp_reserved*/
+    0,                                  /*tp_repr*/
+    0,                                  /*tp_as_number*/
+    0,                                  /*tp_as_sequence*/
+    0,                                  /*tp_as_mapping*/
+    0,                                  /*tp_hash*/
+    0,                                  /*tp_call*/
+    0,                                  /*tp_str*/
+    0,                                  /*tp_getattro*/
+    0,                                  /*tp_setattro*/
+    0,                                  /*tp_as_buffer*/
+    Py_TPFLAGS_DEFAULT,                 /* tp_flags */
+    0,                                  /*tp_doc*/
+    0,                                  /*tp_traverse*/
+    0,                                  /*tp_clear*/
+    0,                                  /*tp_richcompare*/
+    0,                                  /*tp_weaklistoffset*/
+    0,                                  /*tp_iter*/
+    0,                                  /*tp_iternext*/
+    0,                                  /*tp_methods*/
+    0,                                  /* tp_members */
+    0,                                  /* tp_getset */
+    0,                                  /* tp_base */
+    0,                                  /* tp_dict */
+    0,                                  /* tp_descr_get */
+    0,                                  /* tp_descr_set */
+    0,                                  /* tp_dictoffset */
+    0,                                  /* tp_init */
+    PyType_GenericAlloc,                /* tp_alloc */
+    abc_data_new                        /* tp_new */
+};
+
+static int
+_in_weak_set(PyObject *set, PyObject *obj)
+{
+    PyObject *ref;
+    int res;
+    ref = PyWeakref_NewRef(obj, NULL);
+    if (!ref) {
+        if (PyErr_ExceptionMatches(PyExc_TypeError)) {
+            return 0;
+        }
+        return -1;
+    }
+    res = PySet_Contains(set, ref);
+    Py_DECREF(ref);
+    return res;
+}
+
+static _abc_data *
+_get_impl(PyObject *self)
+{
+    PyObject *impl;
+    impl = _PyObject_GetAttrId(self, &PyId__abc_impl);
+    if (!impl) {
+        return NULL;
+    }
+    if (Py_TYPE(impl) != &_abc_data_type) {
+        PyErr_SetString(PyExc_TypeError, "_abc_impl is set to a wrong type");
+        Py_DECREF(impl);
+        return NULL;
+    }
+    return (_abc_data *)impl;
+}
+
+static int
+_in_cache(PyObject *self, PyObject *cls)
+{
+    _abc_data *impl;
+    int res;
+    impl = _get_impl(self);
+    if (!impl) {
+        return -1;
+    }
+    res = _in_weak_set(impl->_abc_cache, cls);
+    Py_DECREF(impl);
+    return res;
+}
+
+static int
+_in_negative_cache(PyObject *self, PyObject *cls)
+{
+    _abc_data *impl;
+    int res;
+    impl = _get_impl(self);
+    if (!impl) {
+        return -1;
+    }
+    res = _in_weak_set(impl->_abc_negative_cache, cls);
```

Comment:
You can declare variables in place: `int res = ` (I always forget about this myself :) )